extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -121,15 +120,16 @@ private void registerNativeImageResources(BuildProducer<NativeImageResourceBuild
             throws IOException, URISyntaxException {
         List<String> resources = new ArrayList<>();
         try {
-            List<String> locations = new ArrayList<>(flywayBuildConfig.locations.orElse(Collections.emptyList()));
-            if (locations.isEmpty()) {
-                locations.add("db/migration");
+            String[] defaultMigrationLocation = { "db/migration" };
+            String[] locations = flywayBuildConfig.locations.orElse(defaultMigrationLocation);
+            if (locations.length == 0) {
+                locations = defaultMigrationLocation;
             }
             // Locations can be a comma separated list
             for (String location : locations) {
                 // Strip any 'classpath:' protocol prefixes because they are assumed
                 // but not recognized by ClassLoader.getResources()
-                if (location != null && location.startsWith(CLASSPATH_APPLICATION_MIGRATIONS_PROTOCOL + ':')) {
+                if (location.startsWith(CLASSPATH_APPLICATION_MIGRATIONS_PROTOCOL + ':')) {
```

Comment:
I think I would keep this null check here just to be safe.

Reply:
I already discussed with @machi1990 that the null check is not necessary. If you want to keep it please clarify it with him as you both reviewers have oposite opinions.

Also the old code was not written "save" but assumes that `location` is never `null`. Line 135 `....getResources(location);` is not protected by the null check above and asserts the parameter to be not null in the first line of the call.